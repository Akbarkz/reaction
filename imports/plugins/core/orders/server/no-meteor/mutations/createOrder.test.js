import Factory, { createFactoryForSchema } from "/imports/test-utils/helpers/factory";
import createOrder, { inputSchema } from "./createOrder";
import mockContext from "/imports/test-utils/helpers/mockContext";

createFactoryForSchema("CreateOrderInput", inputSchema);

test("throws access denied if guest checkout is disabled and no accountId", async () => {
  const shop = Factory.Shop.makeOne({
    isGuestCheckoutAllowed: false
  });
  const input = Factory.CreateOrderInput.makeOne();
  const createPaymentForFulfillmentGroup = () => {};

  mockContext.accountId = null;
  mockContext.collections.Shops.findOne.mockReturnValueOnce(Promise.resolve(shop));

  const promise = createOrder(mockContext, { ...input, createPaymentForFulfillmentGroup });
  return expect(promise).rejects.toThrowErrorMatchingSnapshot();
});
