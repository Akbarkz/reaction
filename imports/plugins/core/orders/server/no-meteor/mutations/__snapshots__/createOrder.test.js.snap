// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws access denied if guest checkout is disabled and no accountId 1`] = `"You must be logged in to place an order"`;
